src/core/directive.rs

```diff
@@ -1,18 +1,24 @@
+use std::collections::HashMap;
+
 use async_graphql::parser::types::ConstDirective;
 use async_graphql::{Name, Pos, Positioned};
 use serde::{Deserialize, Serialize};
 use serde_json::{Map, Value};
 use serde_path_to_error::deserialize;
 
-use crate::core::blueprint;
 use crate::core::valid::{Valid, ValidationError, Validator};
 
 fn pos<A>(a: A) -> Positioned<A> {
     Positioned::new(a, Pos::default())
 }
 
-#[allow(dead_code)]
-fn to_const_directive(directive: &blueprint::Directive) -> Valid<ConstDirective, String> {
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, schemars::JsonSchema)]
+pub struct Directive {
+    pub name: String,
+    pub arguments: HashMap<String, Value>,
+}
```

Comment:
What if we wish to attach additional information such as IR to a directive in future? Something that won't be meaningful in Config but useful in blueprint.

Reply:
In my humble opinion we should keep them separated - the `blueprint::Directive` and `config::Directive`. One change I can see immediately is in `Config` I would like to use a BTreeMap or an IndexMap because I would like to maintain an order, but in Blueprint I could use a HashMap for better performance.

Reply:
applied the suggested change